src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,180 @@
+import {SyntheticEventTarget} from '../../../esl-utils/dom/events/target';
+import {ESLMixinElement} from '../../../esl-mixin-element/ui/esl-mixin-element';
+import {overrideEvent} from '../../../esl-utils/dom/events/misc';
+
+type SwipeDirection = 'left' | 'right' | 'up' | 'down';
+type SwipeEventName = 'swipe' | 'swipe:left' | 'swipe:right' | 'swipe:up' | 'swipe:down';
+
+interface SwipeEvent {
+  direction: SwipeDirection;
+  distanceX: number;
+  distanceY: number;
+  touchStartOriginalEvent: PointerEvent;
+  touchEndOriginalEvent: PointerEvent;
+}
+
+interface SwipeEventTargetConfig {
+  threshold: number;
+  units: string;
+  timeout: number;
+}
+
+export class ESLSwipeEventTarget extends SyntheticEventTarget {
+  protected xDown: number;
+  protected yDown: number;
+  protected timeDown: number;
+  protected startEl: HTMLElement | null;
+  protected startEvent: PointerEvent;
+  protected config: SwipeEventTargetConfig;
+  protected static unitsAvailable = ['vw', 'vh', 'px'];
+  protected static defaultConfig: SwipeEventTargetConfig = {
+    threshold: 20,
+    units: 'px',
+    timeout: 500
+  };
+
+  protected constructor(public readonly target: Element, threshold?: string, timeout?: number) {
+    super();
+    this.config = this.getConfig(threshold, timeout);
+  }
+
+  /**
+   * Passes threshold into number and units, creates config from passed threshold and distance values or uses default ones.
+   * @param threshold - the distance threshold (px, vh, vw supported) that must be traveled before a swipe event is triggered.
+   * @param timeout - the time in milliseconds that must elapse between pointerdown and pointerup events before a wipe event is triggered.
+   * @returns ESLSwipeEventTarget configuration {@link SwipeEventTargetConfig}.
+   */
+  protected getConfig(threshold?: string, timeout?: number): SwipeEventTargetConfig {
+    const config = ESLSwipeEventTarget.defaultConfig;
+
+    if (threshold) {
+      ESLSwipeEventTarget.unitsAvailable.forEach((unit: string) => {
+        if (threshold.indexOf(unit) > 0) {
+          config.units = unit;
+          config.threshold = parseInt(threshold.replace(unit, ''), 10) || ESLSwipeEventTarget.defaultConfig.threshold;
+        }
+      });
+    }
+
+    config.timeout = timeout || config.timeout;
+
+    return config;
+  }
+
+  /**
+   * @param $el - the element to listen for swipe events on
+   * @param threshold - the distance threshold (px, vh, vw supported) that must be traveled before a swipe event is triggered. Optional. Defaults to 20px
+   * @param timeout - the time in milliseconds that must elapse between pointerdown and pointerup events before a wipe event is triggered. Optional.
+   * Defaults to 500.
+   * @returns Returns the instance of ESLSwipeEventTarget {@link ESLSwipeEventTarget}.
+   */
+  public static for($el: Element | ESLMixinElement, threshold?: string, timeout?: number): ESLSwipeEventTarget {
```

Comment:
In case we face the situation when we need more than two setting attributes here we will need to create a config object anyway (but it will be a Breaking change for users)

Recommend considering the following API for users:
```
  ESLSwipeEventTarget.for($el: ESLDomElementTarget, options: Partial<SwipeEventTargetConfig>): ESLSwipeEventTarget {
```

Please also note that we use the ESLDomElementTarget interface here and `resolveDomTarget(target)` to resolve it to the Dom element.
First of all, ESL as a library uses a much more wide definition in API than just Element / ESLMixinElement
Also, under the hood, it does not reach the ESLMixinElement at all due to all ESL modules should have the most independent from other modules' state to maintain optimal tree shaking for every ESL user.